### PR TITLE
Migrate deprecated features in picocli to v4.0

### LINKED ISSUE #2701: Migrate deprecated features in picocli to v4.0
### Description

Some of the features that picocli offers have been deprecated, specifically `parseWithHandlers` has been superseded by CommandLine.execute since 4.0 

Further information can be found [https://picocli.info/#_migration](https://picocli.info/#_migration) on how to migrate

### PR DESCRIPTION
**Pull request by gfukushima**
## PR description
This PR migrates deprecated features in picocli to the recommended for 4.0
- Replaces parseWithHandles by the execute API
- Replaces deprecated interfaces by their equivalent in picocli 4.0

## Fixed Issue(s)

Fixes #2701 
## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/error/BesuExecutionExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.hyperledger.besu.cli.error;
```
Comment (Gabriel-Trintinalia) [inline besu/src/main/java/org/hyperledger/besu/cli/error/BesuExecutionExceptionHandler.java:1]: Copyright

File: besu/src/test/java/org/hyperledger/besu/cli/operator/OperatorSubCommandTest.java [hunk 2]
```diff
@@ -280,34 +294,40 @@ public void shouldGenerateSECP256R1KeysWhenSetAsEcCurve() throws IOException {
...
+                OperatorSubCommand.COMMAND_NAME,
+                OperatorSubCommand.GENERATE_BLOCKCHAIN_CONFIG_SUBCOMMAND_NAME,
+                "--config-file",
+                configFilePath.getPath(),
+                "--to",
+                tmpOutputDirectoryPath.toString())
+            .args(cmd().argsArray())
+            .argsArray());
+    assertThat(commandErrorOutput.toString(UTF_8).trim())
+        .endsWith("Missing config section in config file");
```
Comment (lucassaldanha) [inline besu/src/test/java/org/hyperledger/besu/cli/operator/OperatorSubCommandTest.java:330]: It looks like all these updated tests are doing basically the same thing: asserting that a specific error message is sent to the error output. It is probably worth having a helper method that takes the config file path and the expected error msg as parameters so we don't end up with a lot of duplication.
```

Comment (gfukushima) [reply besu/src/test/java/org/hyperledger/besu/cli/operator/OperatorSubCommandTest.java]: Done, thanks for the feedback

**Review by shemnon — APPROVED** [review]
LGTM

**Pull request merged**